scripts/MeadeCommandParser.py

```diff
@@ -40,75 +135,165 @@ def removeLinePrefix(line):
     raise Exception("Could not locate start and stop of comment block.")
 
 START_LINE = startStop[0]
-END_LINE  = startStop[1]
+END_LINE = startStop[1]
 
 print(("Start and end of block: {0}, {1} ".format(START_LINE, END_LINE)))
 
-familyDividers = []
+family_dividers = []
 for i in range(START_LINE, END_LINE):
     for div in ["//------------------------------------------------------------------", "// --"]:
         if div in content[i]:
-            familyDividers.append(i)
+            family_dividers.append(i)
 
-print(("Found {0} family groups ".format(len(familyDividers))))
+print(("Found {0} family groups ".format(len(family_dividers))))
 
-allCommands = []
-for i in range(len(familyDividers) - 1):
-    start = familyDividers[i]
-    end = familyDividers[i + 1]
+all_commands = []
+for i in range(len(family_dividers) - 1):
+#for i in range(0, 6):
+    start = family_dividers[i] + 1
+    end = family_dividers[i + 1]
+
+    new_family = Family()
+    new_family.name = remove_line_prefix(content[start])
+
+    # Find command groups
+    sub_offsets = list()
+    sub_offsets.append(start+2)
+    for j in range(start+2, end):
+        if content[j] == "//":
+            sub_offsets.append(j)
     
-    newFamily = Family()
-    if "//------------------------------------------------------------------" in content[start]:
-        newFamily.name = removeLinePrefix(content[start + 1])
-    elif "// --" in content[start]:
-        nameCleanup = content[start].replace("// -- ", "")
-        nameCleanup = nameCleanup.replace(" --", "")
-        newFamily.name = nameCleanup
-
-    for y in range(start + 1, end - 1):
-        newCommand = Command()
-
-        # Command
-        if content[y].startswith("// :"):
-            newCommand.command = removeLinePrefix(content[y])
-        
-            # Short Description
-            newCommand.short = removeLinePrefix(content[y + 1])
-            y+=2
-
-            k = y
-            while not content[k].startswith("//      Returns:"):
-                newCommand.long += removeLinePrefix(content[k])
-                k += 1
-            y = k
+    for k in range(0, len(sub_offsets)-1):
+        command = Command()
+        sub_start = sub_offsets[k]
+        sub_end = sub_offsets[k+1]
         
+        for l in range(sub_start, sub_end):
+            if content[l] == "//":
+                continue
+            
+            # Command
+            if content[l].startswith("// :"):
+                command.command = remove_line_prefix(content[l])
+
+            # Description
+            if content[l].startswith("//      Description:"):
```

Comment:
Can we change all these to be something like `.startsWith("//") && .contains("Description:")` ? We don't want to fail because of whitespace...